### PR TITLE
Remove singleprecision

### PR DESCRIPTION
**Pull request by andyspiros**
This pull requests completes this one: C2SM-RCM/serialbox#7
Now merging to the right repository from the right repository. Single-precision and double-precision unittests included for both the C++ implementation and the C wrapper.

### CODE REVIEW

File: unittest/serializer/SerializerUnittest.cpp [hunk 1]
```diff
@@ -128,20 +135,20 @@ TEST_F(SerializerUnittest, Write)
...
-        ASSERT_EQ(sizeof(double)*jsize*ksize, filelength);
+        ASSERT_EQ(sizeof(float)*jsize*ksize, filelength);
 
         std::vector<char> rawdata(filelength);
         fs.read(rawdata.data(), filelength);
         fs.close();
 
         // Check data
-        const double* pNewData = reinterpret_cast<const double*>(rawdata.data());
+        const float* pNewData = reinterpret_cast<const float*>(rawdata.data());
```
Comment (pspoerri) [inline unittest/serializer/SerializerUnittest.cpp:145]: Why the change from double to float here? The data we are reading is double precision right?

Comment (pspoerri) [inline unittest/serializer/SerializerUnittest.cpp:145]: Ah I think I see. I think we should have different tests for single and double precision. I would split the test function and then template it in single and double precision.

**Comment by pspoerri** [comment]
I have one problem with this commit: We mix the tests with single and double precision. I would rather see two separate tests for single and double precision, because this is the most common use case in our code base. 
The mixed types test makes also sense as an additional test case. 

Maybe we can add this as a todo for a later stage?

**Comment by andyspiros** [comment]
Well, we have some tests where we mix them, some others where we have separate tests for them. There is a substantial difference between Serialbox and the usual case with STELLA and the Dycore: Serialbox does inherently support both single and double precision in the same build. For Serialbox there is no clear distinction between single and double precision floats: they are simply two different types, just like double and ints are different types. The serializer should be able (and it is) to handle at the same time both single's and double's. For this to work we better have some cases where we actually write/read in the same place single's and double's. This is why I think the current situation is good.

**Comment by pspoerri** [comment]
Ok, fine. I'm not completely happy with the way the unittests are setup to test and handle this case. I'll create an issue, so that we can keep track of this. 

You can merge.

**Comment by andyspiros** [comment]
Just to make clear the previous status: the unittests already had tests with both single precision and double precision values, e.g. in the metainformation of savepoints and fields. Just some of these tests relied on the `Real` typedef, and for those I made sure we have both `float` and `double` tested.
